### PR TITLE
 #24:art:  feat(Application design): Layout of the Application ref:

### LINKED ISSUE #24: 🎨  [Application design]  Layout of the Application
## Application layout

When a user logs in, they are navigated to the application homepage (Dashboard). Here they can view an analysis of all their bots and all the bots they are building and go to their builder.

---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user I wish to view a navigation pane to other features of the application and a toolbar with my user menu.

### Description

The application layout at the moment looks like below:

Screenshot 2022-08-03 at 15 58 30

From here I can :
1. Interact with the sidenav at the left which contains my navigation.
2. Interact with the user component which contains my user menu.
3. Interact with the application pages which contain the various features of the application.

We have prepared a new design to improve the current UI as shown below:

Screenshot 2022-08-03 at 16 06 09

Your assignment, make it look like that picture. Ignore the sections filled in gray colour for now as those are in another feature.

### Technical Analysis

This original page layout is located in libs/elements/layout/page-convl/src/lib/components. The functionality is spread across the different components in this library.

For this feature you'll have to:
```markdown

1. Edit the convl-sidemenu component to match the new design.
2. Edit the convl-navbar component to match the new design.
```

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the image.
2. The component is well structured following the Technical Analysis

### PR DESCRIPTION
**Pull request by meshack34**
# Description
Reference the issue related to this PR as shown [here](https://github.com/italanta/italanta-apps/issues/24).
### Summary
*Fixes #24* completely
When a user logs in, they are able to navigated to the application homepage (Dashboard). Here they can view an analysis of all their bots and all the bots they are building and go to their builder.

 **Dependencies Added**
- Added font awesome
`npm install --save-dev @fortawesome/fontawesome-free`

**From here one can :**

Interact with the sidenav at the left which contains my navigation.
Interact with the user component which contains my user menu.
Interact with the application pages which contain the various features of the application.
We have prepared a new design to improve the current UI as shown below:

# Video and Screenshot

https://user-images.githubusercontent.com/41685560/186847859-4731385d-2c99-4034-bc29-cf295f56efcd.mp4

#
elewscreen

Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code

### CODE REVIEW

File: apps/conv-learning-manager/src/styles.scss [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
 @import 'https://fonts.googleapis.com/icon?family=Material+Icons';
 
 @import '_variables';
 
 * {
   font-family: 'Poppins', sans-serif;
-}
\ No newline at end of file
+}
+
```
Comment (pmutua) [inline apps/conv-learning-manager/src/styles.scss]: ✅ ok

File: libs/elements/layout/page-convl/src/lib/components/convl-navbar/navbar.component.html [hunk 2]
```diff
@@ -22,6 +22,7 @@
 
       <!-- Default ng-content -->
       <ng-content></ng-content>
+      <app-user-menu></app-user-menu>
 
   </mat-toolbar>
 </nav>
```
Comment (pmutua) [inline libs/elements/layout/page-convl/src/lib/components/convl-navbar/navbar.component.html:28]: ✅ ok

File: libs/elements/layout/page-convl/src/lib/components/convl-navbar/navbar.component.scss [hunk 3]
```diff
@@ -26,10 +26,14 @@
...
   height: 46px;
   padding: 0 12px;
+  
+  
 }
 
 #navbar-title{
-  font-size: 18px;
+  font-size:10px;
+  color: red;
```
Comment (pmutua) [inline libs/elements/layout/page-convl/src/lib/components/convl-navbar/navbar.component.scss:35]: Use theme color **$warn**  
```
#navbar-title{
  font-size:10px;
  color: $warn;
```
The theme file : **libs/elements/theming/theme.scss** helps use to reuse the color in any component.

File: libs/elements/layout/page-convl/src/lib/components/convl-sidemenu/convl-sidemenu.component.html [hunk 4]
```diff
@@ -1,51 +1,64 @@
...
-        </a>
-      </span>
-    </div> -->
+<mat-toolbar class="sidebar-menu">
+
+
+  <div [routerLink]="['/']"
+  mat-list-item class="header top-header" fxLayout="row">
+  
+  <!-- <i matListIcon class="fas fa-columns"></i> -->
```
Comment (pmutua) [inline libs/elements/layout/page-convl/src/lib/components/convl-sidemenu/convl-sidemenu.component.html:10]: This looks fine, but remove  dead code  (commented code) if we are not using it.

File: libs/elements/layout/page-convl/src/lib/components/convl-sidemenu/convl-sidemenu.component.html [hunk 5]
```diff
@@ -1,51 +1,64 @@
...
+  </div>
+  </div>
+  <button mat-icon-button class="sidenavmenu"
+  *ngIf="sidenav.mode === 'over'"
+  (click)="sidenav.toggle()">  
+  <i class="fa fa-bars fa-lg" aria-hidden="true"></i>
+</button>
+  
+  <hr id="menu-header-seperator"/>
+</mat-toolbar>
```
Comment (pmutua) [inline libs/elements/layout/page-convl/src/lib/components/convl-sidemenu/convl-sidemenu.component.html:24]: ✅ok

File: libs/elements/layout/page-convl/src/lib/components/convl-sidemenu/convl-sidemenu.component.html [hunk 6]
```diff
@@ -1,51 +1,64 @@
...
+          </ng-template>
+      
+
+         <div class="category-children" *ngIf="rlaHome.isActive" fxLayout="column" fxLayoutGap="5px">
+          <div  [routerLink]="['/org']"
+               mat-list-item class="child-header child-1"> 
+              <i class='fa-solid fa-gauge-high fa-lg'></i> 
+
+            Dashboard
+          </div>
```
Comment (pmutua) [inline libs/elements/layout/page-convl/src/lib/components/convl-sidemenu/convl-sidemenu.component.html:51]: ✅ ok

File: libs/elements/layout/page-convl/src/lib/components/convl-sidemenu/convl-sidemenu.component.html [hunk 7]
```diff
@@ -1,51 +1,64 @@
...
-      </div>
-    </div>
+    
+
+   </div>
+
+  </mat-sidenav>
+
+</mat-sidenav-container>
+
```
Comment (pmutua) [inline libs/elements/layout/page-convl/src/lib/components/convl-sidemenu/convl-sidemenu.component.html:61]: ✅ ok

File: package.json [hunk 8]
```diff
@@ -50,6 +50,7 @@
     "@angular/cli": "~14.0.0",
     "@angular/compiler-cli": "~14.0.0",
     "@angular/language-service": "~14.0.0",
+    "@fortawesome/fontawesome-free": "^6.1.2",
     "@nrwl/cli": "14.4.2",
     "@nrwl/cypress": "14.4.2",
     "@nrwl/eslint-plugin-nx": "14.4.2",
```
Comment (pmutua) [inline package.json:56]: ✅ ok

**Review by pmutua — CHANGES_REQUESTED** [review]
Good work, we are almost done now before we do the merging there are some few requests I need you to finalize then we will approve this PR.

**Comment by meshack34** [comment]
I have made the necessary changes

**Review by pmutua — APPROVED** [review]

File: libs/elements/layout/page-convl/src/lib/components/convl-navbar/navbar.component.scss [hunk 9]
```diff
@@ -1,3 +1,4 @@
+$warn: warn;
```
Comment (jrosseel) [inline libs/elements/layout/page-convl/src/lib/components/convl-navbar/navbar.component.scss:1]: : "warn" is not valid scss syntax

Comment (jrosseel) [reply libs/elements/layout/page-convl/src/lib/components/convl-navbar/navbar.component.scss:1]: Should refer to a color

File: libs/elements/layout/page-convl/src/lib/components/convl-sidemenu/convl-sidemenu.component.scss [hunk 10]
```diff
@@ -2,33 +2,60 @@
 @import '@angular/material/theming';
 // Import your custom theme
 @import 'libs/elements/theming/common.theme.scss';
+@import '~@fortawesome/fontawesome-free/scss/fontawesome';
+@import '~@fortawesome/fontawesome-free/scss/solid';
+@import url("https://fonts.googleapis.com/icon?family=Material+Icons");
```
Comment (jrosseel) [inline libs/elements/layout/page-convl/src/lib/components/convl-sidemenu/convl-sidemenu.component.scss:7]: @IanOdhiambo9 , check import of url here down in a component

**Review by jrosseel — CHANGES_REQUESTED** [review]
Original logo needs to be put back.

Pending review of @IanOdhiambo9

**Pull request closed without merging**